### PR TITLE
HaplotypeCaller and Mutect2: include reads with deletions in isActive() pileups

### LINKED ISSUE #3830: GATK4 AssemblyRegion traversal does not include reads with deletions in pileups used for isActive() checks
This is a difference vs. GATK3, which did include reads with deletions in the isActive pileups for both HaplotypeCaller and Mutect2 ( @davidbenjamin take note ).

The fix is trivial -- I'll have a patch submitted later today.

### PR DESCRIPTION
**Pull request by droazen**
Previously, reads with deletions at the current loci were not being included in
the pileups passed to isActive(), which was a difference vs. the default settings
in GATK3. This patch changes the default to be to include such reads in both
HaplotypeCaller and Mutect2.

Resolves #3830

### CODE REVIEW

**Comment by droazen** [comment]
@jamesemery please review

File: src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java [hunk 1]
```diff
@@ -305,7 +310,7 @@ public final void traverse() {
      * @param features FeatureManager
      */
     private void processReadShard(Shard<GATKRead> shard, ReferenceDataSource reference, FeatureManager features ) {
-        final Iterator<AssemblyRegion> assemblyRegionIter = new AssemblyRegionIterator(shard, getHeaderForReads(), reference, features, assemblyRegionEvaluator(), minAssemblyRegionSize, maxAssemblyRegionSize, assemblyRegionPadding, activeProbThreshold, maxProbPropagationDistance);
+        final Iterator<AssemblyRegion> assemblyRegionIter = new AssemblyRegionIterator(shard, getHeaderForReads(), reference, features, assemblyRegionEvaluator(), minAssemblyRegionSize, maxAssemblyRegionSize, assemblyRegionPadding, activeProbThreshold, maxProbPropagationDistance, includeReadsWithDeletionsInIsActivePileups());
```
Comment (jamesemery) [inline src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java]: Perhaps you could add a test at the `AssemblyRegionIterator` that enforces that this variable is making a difference? It looks like the `LocusIteratorByState` has tests with both values set but its unclear that its behavior is enforced.

File: src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.testGVCFMode.gatk4.alleleSpecific.g.vcf [hunk 2]
```diff
@@ -239,7 +239,7 @@
...
 20	10004726	.	T	<NON_REF>	.	.	END=10004768	GT:DP:GQ:MIN_DP:PL	0/0:67:99:63:0,105,1575
-20	10004769	.	TAAAACTATGC	T,<NON_REF>	981.73	.	AS_RAW_BaseQRankSum=|1.6,1|NaN;AS_RAW_MQ=133200.00|32185.00|0.00;AS_RAW_MQRankSum=|-6.7,1|NaN;AS_RAW_ReadPosRankSum=|2.2,1|NaN;AS_SB_TABLE=14,23|7,20|0,0;BaseQRankSum=1.603;ClippingRankSum=0.000;DP=80;ExcessHet=3.0103;MLEAC=1,0;MLEAF=0.500,0.00;MQRankSum=-6.625;RAW_MQ=217467.00;ReadPosRankSum=2.215	GT:AD:DP:GQ:PL:SB	0/1:37,27,0:64:99:1019,0,1470,1130,1554,2684:14,23,7,20
+20	10004769	.	TAAAACTATGC	T,<NON_REF>	981.73	.	AS_RAW_BaseQRankSum=|1.6,1|NaN;AS_RAW_MQ=133200.00|32185.00|0.00;AS_RAW_MQRankSum=|-6.7,1|NaN;AS_RAW_ReadPosRankSum=|2.2,1|NaN;AS_SB_TABLE=14,23|7,20|0,0;BaseQRankSum=1.603;ClippingRankSum=0.000;DP=79;ExcessHet=3.0103;MLEAC=1,0;MLEAF=0.500,0.00;MQRankSum=-6.625;RAW_MQ=216626.00;ReadPosRankSum=2.215	GT:AD:DP:GQ:PL:SB	0/1:37,27,0:64:99:1019,0,1471,1130,1554,2684:14,23,7,20
```
Comment (jamesemery) [inline src/test/resources/org/broadinstitute/hellbender/tools/haplotypecaller/expected.testGVCFMode.gatk4.alleleSpecific.g.vcf]: These look promising

File: src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java (hunk 1 above)
Comment (droazen) [reply src/main/java/org/broadinstitute/hellbender/engine/AssemblyRegionWalker.java]: I've added a unit test for `AssemblyRegionIterator` that proves that this feature causes reads with deletions to be retained in the pileups sent to the `AssemblyRegionEvaluator`.

File: src/test/java/org/broadinstitute/hellbender/engine/AssemblyRegionIteratorUnitTest.java [hunk 3]
```diff
@@ -112,4 +114,68 @@ public void testRegionsHaveCorrectReadsAndSize( final String reads, final String
...
+
+    @DataProvider
+    public Object[][] testIncludeReadsWithDeletionsInIsActivePileupsData() {
+        return new Object[][] {
+                { NA12878_20_21_WGS_bam, b37_reference_20_21, new SimpleInterval("20", 10004770, 10004770), true, 29 },
+                { NA12878_20_21_WGS_bam, b37_reference_20_21, new SimpleInterval("20", 10004770, 10004770), false, 0 }
+        };
+    }
+
+    @Test(dataProvider = "testIncludeReadsWithDeletionsInIsActivePileupsData")
```
Comment (jamesemery) [inline src/test/java/org/broadinstitute/hellbender/engine/AssemblyRegionIteratorUnitTest.java]: That is one complicated test, It looks good though!

**Pull request merged**